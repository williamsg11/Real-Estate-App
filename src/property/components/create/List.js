/**
 * @flow
 */
import React, {Component, PropTypes} from 'react';
import {
  ListView,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import colors from '../../../common/colors';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Separator from './../../../components/Separator';

export default class List extends Component {
  static propTypes = {
    field: PropTypes.string.isRequired,
    updateListing: PropTypes.func.isRequired,
    collection: PropTypes.array.isRequired,
    selected: PropTypes.string,
  };

  renderRow = item => {
    const {updateListing, field, selected} = this.props;
    return (
      <View key={item}>
        <TouchableHighlight
          onPress={() => updateListing(field, item)}
          underlayColor="transparent">
          <View style={styles.row}>
            <Text
              style={[
                styles.title,
                selected === item && {color: colors.accent},
              ]}>
              {item}
            </Text>
            <Ionicons
              name="ios-arrow-forward"
              color={colors.smokeGreyLight}
              size={30}
            />
          </View>
        </TouchableHighlight>
        <Separator />
      </View>
    );
  };

  render() {
    const {collection, header} = this.props;

    let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
    let dataSource = ds.cloneWithRows(collection);

    return (
      <View style={styles.container}>

        {header}
        <View style={styles.menuContainer}>
          <ListView
            dataSource={dataSource}
            style={styles.list}
            enableEmptySections={true}
            renderRow={this.renderRow}
            automaticallyAdjustContentInsets={false}
            showsVerticalScrollIndicator={false}
            contentInset={{bottom: 50}}
            removeClippedSubviews={false}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  menuContainer: {
    justifyContent: 'flex-end',
  },
  descriptionContainer: {
    flex: 1,
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  description: {
    color: colors.darkGrey,
    fontWeight: '600',
    fontSize: 20,
    padding: 30,
    textAlign: 'center',
  },
  list: {
    padding: 10,
    backgroundColor: 'white',
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 20,
    paddingBottom: 20,
  },
  title: {
    flex: 1,
    color: 'black',
    fontWeight: '600',
    fontSize: 16,
  },
  stage: {
    color: colors.darkGrey,
  },
});
