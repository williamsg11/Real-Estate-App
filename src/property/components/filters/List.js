/**
 * @flow
 */
import React, {Component, PropTypes} from 'react';
import {
  ListView,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import colors from '../../../common/colors';
import {CountryPropType} from './../../common/proptypes';

export default class List extends Component {
  static propTypes = {
    selected: PropTypes.string.isRequired,
    onSelect: PropTypes.func.isRequired,
    ranges: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    hint: PropTypes.string,
  };

  constructor() {
    super();
    this.renderRow = this.renderRow.bind(this);
  }

  renderRow(item) {
    const {onSelect, selected, hint} = this.props;
    return (
      <View style={styles.row}>
        <TouchableHighlight
          hitSlop={{top: 20, left: 20, right: 20, bottom: 20}}
          onPress={() => {
            onSelect(item);
          }}
          underlayColor="transparent">
          <Text
            style={[
              styles.price,
              selected === item && {color: colors.tomato, fontWeight: '500'},
            ]}>
            {item}

            <Text style={styles.hint}>
              {item !== 'Any' && hint && ` ${hint}`}
            </Text>

          </Text>
        </TouchableHighlight>
      </View>
    );
  }

  render() {
    const {ranges, title, titleStyle} = this.props;
    let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    let dataSource = ds.cloneWithRows(ranges);

    return (
      <View style={styles.container}>
        <Text style={[styles.title, titleStyle]}>{title}</Text>
        <ListView
          dataSource={dataSource}
          contentContainerStyle={styles.contentContainer}
          style={styles.listStyle}
          horizontal={true}
          enableEmptySections={true}
          renderRow={this.renderRow}
          automaticallyAdjustContentInsets={false}
          showsHorizontalScrollIndicator={false}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },
  contentContainer: {
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingVertical: 10,
  },
  row: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 10,
    paddingHorizontal: 20,
  },
  listStyle: {
    flex: 1,
    flexDirection: 'row',
  },
  title: {
    justifyContent: 'center',
    textAlign: 'center',
    fontWeight: '600',
    fontSize: 14,
  },
  price: {
    fontSize: 14,
  },
  hint: {
    fontSize: 12,
  },
});
