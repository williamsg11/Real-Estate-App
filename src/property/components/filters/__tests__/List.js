import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import List from './../List';

test('renders AddressPicker', () => {
  const tree = renderer
    .create(
      <List
        selected="test"
        onSelect={() => {}}
        ranges={['10-20']}
        title="test"
        field="type"
        updateListing={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
