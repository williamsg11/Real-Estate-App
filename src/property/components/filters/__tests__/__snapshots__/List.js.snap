// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddressPicker 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "600",
          "justifyContent": "center",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
  <RCTScrollView
    alwaysBounceHorizontal={true}
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "paddingVertical": 10,
      }
    }
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    enableEmptySections={true}
    horizontal={true}
    onContentSizeChange={null}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={undefined}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    sendMomentumEvents={false}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }
  >
    <RCTScrollContentView
      collapsable={false}
      removeClippedSubviews={undefined}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "paddingVertical": 10,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 14,
                },
                false,
              ]
            }
          >
            10-20
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                }
              }
            />
          </Text>
        </View>
      </View>
    </RCTScrollContentView>
  </RCTScrollView>
</View>
`;
